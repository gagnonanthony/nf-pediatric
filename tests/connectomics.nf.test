nextflow_pipeline {

    options "-stub-run"

    name "Test nf-pediatric profile connectomics"
    script "../main.nf"

    test("Connectomics profile with metrics - should run successfully") {

        when {
            params {

                params.input = "$projectDir/tests/data/samplesheet_testconn.csv"
                params.outdir = "test_results/"

                params.connectomics = true

            }
        }

        then {
            assert workflow.success
            assert workflow.trace.tasks().size() == 7
        }

    }

    test("Connectomics profile without metrics - should run successfully") {

        when {
            params {

                params.input = "$projectDir/tests/data/samplesheet_testconn_nometrics.csv"
                params.outdir = "test_results/"

                params.connectomics = true

            }
        }

        then {
            assert workflow.success
            assert workflow.trace.tasks().size() == 7
        }

    }

    test("Connectomics profile without trk - should fail") {

        when {
            params {

                params.input = "$projectDir/tests/data/samplesheet_testconn_notrk.csv"
                params.outdir = "test_results/"

                params.connectomics = true

            }
        }

        then {
            assert workflow.failed
            assert workflow.stdout.contains("Please provide a trk file for sample: sub-test")
        }

    }

    test("Connectomics + infant profile - should run successfully") {

        when {
            params {

                params.input = "$projectDir/tests/data/samplesheet_testconn_infant.csv"
                params.outdir = "test_results/"

                params.connectomics = true
                params.infant = true

            }
        }

        then {
            assert workflow.success
            assert workflow.trace.tasks().size() == 7
        }

    }

}
