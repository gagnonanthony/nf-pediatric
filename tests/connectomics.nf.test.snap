{
    "Connectomics profile with metrics - should run successfully": {
        "content": [
            10,
            {
                "CONNECTIVITY_AFDFIXEL": {
                    "scilpy": 1.6
                },
                "CONNECTIVITY_METRICS": {
                    "scilpy": "2.0.2"
                },
                "CONNECTIVITY_VISUALIZE": {
                    "scilpy": "2.0.2"
                },
                "FILTERING_COMMIT": {
                    "scilpy": 1.6
                },
                "TRACTOGRAM_DECOMPOSE": {
                    "scilpy": "2.0.2"
                },
                "TRANSFORM_LABELS": {
                    "ants": "2.4.3"
                },
                "Workflow": {
                    "nf/pediatric": "v1.0dev"
                }
            },
            [
                "multiqc",
                "multiqc/global_multiqc_data",
                "multiqc/global_multiqc_plots",
                "multiqc/global_multiqc_report.html",
                "pipeline_info",
                "pipeline_info/pipeline_software_mqc_versions.yml",
                "sub-test",
                "sub-test/anat",
                "sub-test/anat/sub-test_space-DWI_seg-BrainnetomeChild_dseg.nii.gz",
                "sub-test/dwi",
                "sub-test/dwi/sub-test_desc-filtered_tracking.h5",
                "sub-test/dwi/sub-test_desc-preproc_tracking.h5",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-len.npy",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-len.png",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-sc.npy",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-sc.png",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-vol.npy",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-vol.png",
                "sub-test/figures",
                "sub-test/figures/sub-test_gradients_mqc.png",
                "sub-test/figures/sub-test_labels_mqc.png",
                "sub-test/multiqc",
                "sub-test/multiqc/sub-test_multiqc_data",
                "sub-test/multiqc/sub-test_multiqc_plots",
                "sub-test/multiqc/sub-test_multiqc_report.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-01-02T23:24:43.348507"
    },
    "Connectomics profile without metrics - should run successfully": {
        "content": [
            10,
            {
                "CONNECTIVITY_AFDFIXEL": {
                    "scilpy": 1.6
                },
                "CONNECTIVITY_METRICS": {
                    "scilpy": "2.0.2"
                },
                "CONNECTIVITY_VISUALIZE": {
                    "scilpy": "2.0.2"
                },
                "FILTERING_COMMIT": {
                    "scilpy": 1.6
                },
                "TRACTOGRAM_DECOMPOSE": {
                    "scilpy": "2.0.2"
                },
                "TRANSFORM_LABELS": {
                    "ants": "2.4.3"
                },
                "Workflow": {
                    "nf/pediatric": "v1.0dev"
                }
            },
            [
                "multiqc",
                "multiqc/global_multiqc_data",
                "multiqc/global_multiqc_plots",
                "multiqc/global_multiqc_report.html",
                "pipeline_info",
                "pipeline_info/pipeline_software_mqc_versions.yml",
                "sub-test",
                "sub-test/anat",
                "sub-test/anat/sub-test_space-DWI_seg-BrainnetomeChild_dseg.nii.gz",
                "sub-test/dwi",
                "sub-test/dwi/sub-test_desc-filtered_tracking.h5",
                "sub-test/dwi/sub-test_desc-preproc_tracking.h5",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-len.npy",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-len.png",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-sc.npy",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-sc.png",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-vol.npy",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-vol.png",
                "sub-test/figures",
                "sub-test/figures/sub-test_gradients_mqc.png",
                "sub-test/figures/sub-test_labels_mqc.png",
                "sub-test/multiqc",
                "sub-test/multiqc/sub-test_multiqc_data",
                "sub-test/multiqc/sub-test_multiqc_plots",
                "sub-test/multiqc/sub-test_multiqc_report.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-01-02T23:25:39.287679"
    },
    "Connectomics + infant profile - should run successfully": {
        "content": [
            10,
            {
                "CONNECTIVITY_AFDFIXEL": {
                    "scilpy": 1.6
                },
                "CONNECTIVITY_METRICS": {
                    "scilpy": "2.0.2"
                },
                "CONNECTIVITY_VISUALIZE": {
                    "scilpy": "2.0.2"
                },
                "FILTERING_COMMIT": {
                    "scilpy": 1.6
                },
                "TRACTOGRAM_DECOMPOSE": {
                    "scilpy": "2.0.2"
                },
                "TRANSFORM_LABELS": {
                    "ants": "2.4.3"
                },
                "Workflow": {
                    "nf/pediatric": "v1.0dev"
                }
            },
            [
                "multiqc",
                "multiqc/global_multiqc_data",
                "multiqc/global_multiqc_plots",
                "multiqc/global_multiqc_report.html",
                "pipeline_info",
                "pipeline_info/pipeline_software_mqc_versions.yml",
                "sub-test",
                "sub-test/anat",
                "sub-test/anat/sub-test_space-DWI_seg-BrainnetomeChild_dseg.nii.gz",
                "sub-test/dwi",
                "sub-test/dwi/sub-test_desc-filtered_tracking.h5",
                "sub-test/dwi/sub-test_desc-preproc_tracking.h5",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-len.npy",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-len.png",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-sc.npy",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-sc.png",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-vol.npy",
                "sub-test/dwi/sub-test_seg-BrainnetomeChild_stat-vol.png",
                "sub-test/figures",
                "sub-test/figures/sub-test_gradients_mqc.png",
                "sub-test/figures/sub-test_labels_mqc.png",
                "sub-test/multiqc",
                "sub-test/multiqc/sub-test_multiqc_data",
                "sub-test/multiqc/sub-test_multiqc_plots",
                "sub-test/multiqc/sub-test_multiqc_report.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-01-02T23:26:54.24494"
    }
}